pkg/kubectl/service.go

```diff
@@ -183,8 +183,16 @@ func generate(genericParams map[string]interface{}) (runtime.Object, error) {
 	if len(params["type"]) != 0 {
 		service.Spec.Type = api.ServiceType(params["type"])
 	}
-	if service.Spec.Type == api.ServiceTypeLoadBalancer {
+	switch service.Spec.Type {
+	case api.ServiceTypeNodePort:
+		for i, port := range service.Spec.Ports {
+			service.Spec.Ports[i].NodePort = port.Port
+		}
+	case api.ServiceTypeLoadBalancer:
 		service.Spec.LoadBalancerIP = params["load-balancer-ip"]
+	case "Headless":
```

Comment:
While type==Headless seems semantically sane, I think it will encourage people to try that in the YAML/JSON (regardless of your warnings in the docs).  I think we should just say `--clusterIP=None` creates a headless Service, or else we should actually add support for `type=Headless`.